linkerd/http-route/src/http/match/query_param.rs

```diff
@@ -0,0 +1,96 @@
+use http::uri::Uri;
+use regex::Regex;
+
+#[derive(Clone, Debug)]
+pub enum MatchQueryParam {
+    Exact(String, String),
+    Regex(String, Regex),
+}
+
+// === impl MatchQueryParam ===
+
+impl MatchQueryParam {
+    pub fn is_match(&self, uri: &Uri) -> bool {
+        uri.query().map_or(false, |qs| {
+            url::form_urlencoded::parse(qs.as_bytes()).any(|(q, p)| match self {
+                Self::Exact(n, v) => *n == *q && *v == *p,
+                Self::Regex(n, r) => {
+                    if *n == *q {
+                        if let Some(m) = r.find(&*p) {
+                            // Check that the regex is anchored at the start and
+                            // end of the value.
+                            return m.start() == 0 && m.end() == p.len();
+                        }
```

Comment:
similarly, it would be nice IMO if we ensured all regexes are anchored and then used `Regex::is_match`...not a blocker, though.